official/vision/beta/projects/yolo/ops/preprocessing_ops.py

```diff
@@ -0,0 +1,931 @@
+  """
+
+  area_thr = tf.math.abs(area_thr)
+
+  # Get the scaled and shifted heights of the original
+  # unclipped boxes.
+  og_height = tf.maximum(box_history[:, 2] - box_history[:, 0], 0.0)
+  og_width = tf.maximum(box_history[:, 3] - box_history[:, 1], 0.0)
+
+  # Get the scaled and shifted heights of the clipped boxes.
+  clipped_height = tf.maximum(clipped_boxes[:, 2] - clipped_boxes[:, 0], 0.0)
+  clipped_width = tf.maximum(clipped_boxes[:, 3] - clipped_boxes[:, 1], 0.0)
+
+  # Determine the aspect ratio of the clipped boxes.
+  ar = tf.maximum(clipped_width / (clipped_height + 1e-16),
+                  clipped_height / (clipped_width + 1e-16))
+
+  # Ensure the clipped width adn height are larger than a preset threshold.
+  conda = clipped_width > wh_thr
+  condb = clipped_height > wh_thr
+
+  # Ensure the area of the clipped box is larger than the area threshold.
+  area = (clipped_height * clipped_width) / (og_width * og_height + 1e-16)
+  condc = area > area_thr
+
+  # Ensure the aspect ratio is not too extreme.
+  condd = ar < ar_thr
+
+  cond = tf.expand_dims(
+      tf.logical_and(
+          tf.logical_and(conda, condb), tf.logical_and(condc, condd)),
+      axis=-1)
+
+  # Set all the boxes that fail the test to be equal to zero.
+  indices = tf.where(cond)
+  return indices[:, 0]
+
+
+def resize_and_crop_boxes(boxes, image_scale, output_size, offset, box_history):
+  """Resizes and crops the boxes.
+ 
+  Args:
+    boxes: A `Tensor` for the boxes.
+    image_scale: A `Tensor` for the scaling factor of the image.
+    output_size: A `list` of two integers, a two-element vector or a tensor such
+      that all but the last dimensions are `broadcastable` to `boxes`. The last
+      dimension is 2, which represents [height, width].
+    offset: A `Tensor` for how much translation was applied to the image.
+    box_history: A `Tensor` for the boxes history, which are the boxes that
+      undergo the same augmentations as `boxes`, but no clipping was applied.
+      We can keep track of how much changes are done to the boxes by keeping
+      track of this tensor.
+  
+  Returns:
+    clipped_boxes: A `Tensor` representing the augmented boxes.
+    box_history: A `Tensor` representing the augmented box_history.
+  """
+
+  # Shift and scale the input boxes.
+  boxes *= tf.tile(tf.expand_dims(image_scale, axis=0), [1, 2])
+  boxes -= tf.tile(tf.expand_dims(offset, axis=0), [1, 2])
+
+  # Check the hitory of the boxes.
+  box_history *= tf.tile(tf.expand_dims(image_scale, axis=0), [1, 2])
+  box_history -= tf.tile(tf.expand_dims(offset, axis=0), [1, 2])
+
+  # Clip the shifted and scaled boxes.
+  clipped_boxes = bbox_ops.clip_boxes(boxes, output_size)
+  return clipped_boxes, box_history
+
+
+def apply_infos(boxes,
+                infos,
+                affine=None,
+                shuffle_boxes=False,
+                area_thresh=0.1,
+                seed=None,
+                augment=True):
+  """Clips and cleans the boxes.
```

Comment:
apply_infos is very vague name. Either use better name, or split that into smaller functions.

Reply:
addressed